switchtec.c

```diff
@@ -984,8 +989,9 @@ static long switchtec_dev_ioctl(struct file *filp, unsigned int cmd,
 	case SWITCHTEC_IOCTL_FLASH_PART_INFO:
 		rc = ioctl_flash_part_info(stdev, argp);
 		break;
-	case SWITCHTEC_IOCTL_EVENT_SUMMARY:
-		rc = ioctl_event_summary(stdev, stuser, argp);
+	case SWITCHTEC_IOCTL_EVENT_SUMMARY_LEGACY:
+		rc = ioctl_event_summary(stdev, stuser, argp,
+					 _IOC_SIZE(SWITCHTEC_IOCTL_EVENT_SUMMARY_LEGACY));
```

Comment:
It would be better if you used sizeof(struct switchtec_ioctl_event_summary_legacy) instead of _IOC_SIZE... 
 
Generally macros with an underscore prefix should not be used.

Same with the instance below.